Retarus/retarus_modules/retarus_events_forwarder.py

```diff
@@ -0,0 +1,90 @@
+from threading import Event, Thread
+
+import dateutil.parser
+from kafka import KafkaProducer
+from kafka.errors import KafkaTimeoutError
+from sekoia_automation.connector import Connector
+
+from retarus_modules.metrics import OUTCOMING_EVENTS
+
+
+class RetararusEventsForwarder(Thread):
+    def __init__(
+        self,
+        connector: Connector,
+        queue: SimpleQueue,
+        queue_get_limit: int = 10,
+        queue_get_timeout: float = 1.0,
+        queue_get_block: bool = True,
+        queue_get_retries: int = 10,
+    ):
+        super().__init__()
+        self.connector = connector
+        self.configuration = self.connector.configuration
+        self.kafka_producer = KafkaProducer(bootstrap_servers=self.configuration.kafka_url)
+        self.queue = queue
+        self._stop_event = Event()
+
+        self.queue_get_limit = queue_get_limit
+        self.queue_get_timeout = queue_get_timeout
+        self.queue_get_block = queue_get_block
+        self.queue_get_retries = queue_get_retries
+
+    @property
+    def is_running(self):
+        return not self._stop_event.is_set()
+
+    def stop(self):
+        self._stop_event.set()
+
+    def send_message(self, *messages):
+        try:
+            for message in messages:
+                stub = {
+                    "@timestamp": dateutil.parser.parse(message["ts"]).isoformat(),
+                    "customer": {"intake_key": self.configuration.intake_key},
+                    "message": message,
+                    "event": {
+                        "id": str(uuid.uuid4()),
+                        "created": datetime.now(timezone.utc).isoformat(),
+                        "outcome": "failure",
+                    },
+                }
+                self.kafka_producer.send(
+                    topic=self.configuration.kafka_topic,
+                    value=stub,
+                )
+            self.kafka_producer.flush()
+        except KafkaTimeoutError as ex:
+            self.connector.log_exception(ex, message="Error sending the log to Kafka")
+
+    def run(self):
+        while self.is_running:
+            try:
+                events = self._queue_get_batch()
+                if len(events) > 0:
+                    self.connector.log(
+                        message=f"Forward an event to the intake",
+                        level="info",
+                    )
+                    OUTCOMING_EVENTS.labels(intake_key=self.configuration.intake_key).inc()
+                    self.send_message(*events)
+            except Exception as ex:
+                self.connector.log_exception(ex, message="Failed to forward event")
+
+    def _queue_get_batch(self):
+        result: list[dict] = []
+        i: int = 0
+        while len(result) < self.queue_get_limit and i < self.queue_get_retries:
+            try:
+                result.append(self.queue.get(block=self.queue_get_block, timeout=self.queue_get_timeout))
+                time.sleep(0.5)
```

Comment:
Why sleep here since you already block when getting events from the queue with a timeout ?